simtools/simtel/simtel_histograms.py

```diff
@@ -120,28 +116,45 @@ def _check_consistency(self, first_hist_file, second_hist_file):
                 self._logger.error(msg)
                 raise InconsistentHistogramFormat(msg)
 
-    def combine_histogram_files(self):
+    @property
+    def combined_hists(self):
         """Add the values of the same type of histogram from the various lists into a single
         histogram list."""
         # Processing and combining histograms from multiple files
-        self.combined_hists = []
-        n_files = 0
-        for hists_one_file in self.list_of_histograms:
-            count_file = True
+        if self._combined_hists is None:
+            self._combined_hists = []
+            n_files = 0
+            for hists_one_file in self.list_of_histograms:
+                count_file = True
 
-            if len(self.combined_hists) == 0:
-                # First file
-                self.combined_hists = copy.copy(hists_one_file)
+                if len(self._combined_hists) == 0:
+                    # First file
+                    self._combined_hists = copy.copy(hists_one_file)
 
-            else:
-                for hist, this_combined_hist in zip(hists_one_file, self.combined_hists):
-                    self._check_consistency(hist, this_combined_hist)
+                else:
+                    for hist, this_combined_hist in zip(hists_one_file, self._combined_hists):
+                        self._check_consistency(hist, this_combined_hist)
 
-                    this_combined_hist["data"] = np.add(this_combined_hist["data"], hist["data"])
+                        this_combined_hist["data"] = np.add(
+                            this_combined_hist["data"], hist["data"]
+                        )
 
-            n_files += int(count_file)
+                n_files += int(count_file)
 
-        self._logger.debug(f"End of reading {n_files} files")
+            self._logger.debug(f"End of reading {n_files} files")
+        return self._combined_hists
+
+    @combined_hists.setter
+    def combined_hists(self, new_combined_hists):
+        """
+        Setter for combined_hists.
+
+        Parameters
+        ----------
+        new_combined_hists:
+            Combined histograms.
+        """
+        self._combined_hists = new_combined_hists
```

Comment:
This seems to be not used somewhere in the code and codecov complains about missing test. This setter might need additional checks to only allow actual combined histograms in the right format.

Reply:
I only added this setter to be able to test an invalid format in the unit tests, which is necessary since we dont have invalid histograms to start with. I added a line to test the setter though